components/automate-ui/src/app/page-components/services-sidebar/services-sidebar.component.spec.ts

```diff
@@ -0,0 +1,232 @@
+          release: 'custom/nginx/1.0.1/20190115184823',
+          group: 'nginx.default',
+          status: 'ok',
+          health_check: 'CRITICAL',
+          applications: 'bldr-cache',
+          environment: 'qa',
+          fqdn: '8ouxxga6.example.net',
+          site: 'testsite',
+          channel: 'stable',
+          update_strategy: 'AT-ONCE',
+          previous_health_check: 'NONE',
+          current_health_since: '23 hours 54 minutes',
+          health_updated_at: new Date(),
+          health_check_result: {
+            stdout: 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF…FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF', stderr: 'FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF…FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF', exit_status: 2
+          }
+        }
+      ];
+
+      component.servicesList = mockServices;
+    });
+
+    it('should add the id of the selected service to a list', () => {
+      expect(component.checkedServices.length).toEqual(0);
+
+      component.handleToggleCheckbox(300, true);
+      component.handleToggleCheckbox(400, true);
+
+      expect(component.checkedServices.length).toEqual(2);
+      expect(component.checkedServices).toContain(300 && 400);
+    });
+
+    it('should remove already selected services from the list', () => {
+      component.checkedServices = [300, 400, 500];
+      expect(component.checkedServices.length).toEqual(3);
+
+      component.handleToggleCheckbox(300, false);
+      expect(component.checkedServices.length).toEqual(2);
+      expect(component.checkedServices).not.toContain(300);
+
+      component.handleToggleCheckbox(500, false);
+      expect(component.checkedServices.length).toEqual(1);
+      expect(component.checkedServices).not.toContain(300 && 500);
+    });
+
+    it('should update the count of services selected', () => {
+      component.ngOnInit();
+      expect(component.checkedServicesDisplay).toBe('Services');
+      component.handleToggleCheckbox(300, true);
+      component.handleToggleCheckbox(400, true);
+      expect(component.checkedServices.length).toEqual(2);
+
+      expect(component.checkedServicesDisplay).toBe('2 Services');
+
+      component.handleToggleCheckbox(300, false);
+      expect(component.checkedServices.length).toEqual(1);
+      expect(component.checkedServicesDisplay).toBe('1 Service');
+    });
+
+    xit('should disable the Delete Services button when none are selected', () => {
+      expect(component.checkedServices.length).toEqual(0);
+
+      const deleteButton = fixture.debugElement.query(By.css('#delete-services-button'));
+      expect(deleteButton).not.toBeNull();
+      expect(deleteButton.nativeElement.disabled).toBeTruthy();
+
+      component.handleToggleCheckbox(300, true);
+      expect(component.checkedServices.length).toEqual(1);
+      fixture.detectChanges();
+
+      expect(deleteButton.nativeElement.disabled).not.toBeTruthy();
+    });
```

Comment:
I've been unable to grab a button for selecting - maybe this is because its a stencil component?